packages/fury-adapter-swagger/test/schema-test.js

```diff
@@ -943,4 +944,79 @@ describe('JSON Schema to Data Structure', () => {
     expect(dataStructure.content).to.be.instanceof(StringElement);
     expect(dataStructure.content.attributes.getValue('typeAttributes')).to.deep.equal(['nullable']);
   });
+
+  describe('referencing', () => {
+    it('produces element from reference', () => {
+      const schema = { $ref: '#/definitions/User' };
+
+      const root = {
+        definitions: {
+          User: {
+            type: 'object',
+          },
+        },
+      };
+
+      const dataStructure = schemaToDataStructure(schema, root);
+
+      expect(dataStructure.element).to.equal('dataStructure');
+      expect(dataStructure.content).to.be.instanceof(Element);
+      expect(dataStructure.content.element).to.equal('definitions/User');
+    });
+
+    it('produces element from object containing reference', () => {
+      const schema = {
+        type: 'object',
+        properties: {
+          user: { $ref: '#/definitions/User' },
+        },
+      };
+
+      const root = {
+        definitions: {
+          User: {
+            type: 'object',
+          },
+        },
+      };
+
+      const dataStructure = schemaToDataStructure(schema, root);
+
+      expect(dataStructure.element).to.equal('dataStructure');
+      expect(dataStructure.content).to.be.instanceof(ObjectElement);
+
+      const user = dataStructure.content.get('user');
+      expect(user.element).to.equal('definitions/User');
+    });
+
+    it('produces element from object containing reference to another element property', () => {
```

Comment:
The following test is for the new behaviour, the other two tests are increasing related test coverage.